sdk/textanalytics/Azure.AI.TextAnalytics/tests/AnalyzeHealthcareEntitiesTests.cs

```diff
@@ -307,6 +305,8 @@ public async Task AnalyzeHealthcareEntitiesBatchConvenienceTest()
         }
 
         [RecordedTest]
+        // TODO: https://github.com/Azure/azure-sdk-for-net/issues/31978.
+        [ServiceVersion(Min = TextAnalyticsClientOptions.ServiceVersion.V2022_10_01_Preview)]
```

Comment:
I think it's good to track the issue, but you could set `showStats` based on the client versioned passed in. If your base class doesn't already save the `ServiceVersion` under test, you can do so as a protected member and then conditionally set `showStats` based on that. You only need to worry about the latest version with recorded tests; v3.1 will only run under nightly, so you don't have to worry about recordings being wrong.

The fact that you caught this regression proves testing `showStats` is worth covering rather than disabling entirely.

Reply:
I made the change. With this, I can 1) keep checking for request-level stats in version 2022-05-01 (only document-level stats are broken in this version), and 2) keep the test enabled for version 3.0 (which was not affected by the regression).